lib/CliProxy.js

```diff
@@ -0,0 +1,206 @@
+    request(params, (err, res, body) => {
+      if (err) cb(err)
+      if (!CliProxy._objectPathExists(res, 'body', 'cmds', 0, 'p', 'value') ||
+          !CliProxy._objectPathExists(res, 'headers', 'sg-device-id')) {
+        return cb(new Error('No response commands or no deviceId in header from rapid'))
+      }
+      cb(null, {
+        sessionId: res.body.cmds[0].p.value,
+        deviceId: res.headers['sg-device-id']
+      })
+    })
+  }
+
+  /**
+   * Starts a new pipelineserver
+   * @param port
+   * @param rapidUrl
+   * @param sessionId
+   * @param deviceId
+   * @param shopNumber
+   * @param name
+   * @param cb
+   * @return {Promise}
+   */
+  _startPipelineServer (port, rapidUrl, sessionId, deviceId, shopNumber, name, cb) {
+    const server = restify.createServer()
+
+    server.use(restify.plugins.bodyParser({mapParams: false}))
+
+    server.get('/status', (req, res) => res.send({status: 'is running'}))
+    server.post('/pipelines/.*', this._getPipelineHandlerFunction(rapidUrl, sessionId, deviceId, shopNumber, false))
+    server.post('/trustedPipelines/.*', this._getPipelineHandlerFunction(rapidUrl, sessionId, deviceId, shopNumber, true))
+
+    server.listen(port, () => {
+      logger.info(`${name} proxy is listening on ${port}`)
+      cb(null, server)
+    })
+  }
+
+  _getPipelineHandlerFunction (rapidUrl, sessionId, deviceId, shopNumber, trusted) {
+    /**
+     * Forwards the request to the new rapid
+     * @param {Object} req The request object.
+     * @param {Object} res The response object.
+     */
+    return (req, res, next) => {
+      const cmd = Object.assign({}, pipelineCommand)
+      if (trusted) {
+        cmd.p.type = 'trusted'
+        cmd.p.name = req.url.replace('/trustedPipelines/', '')
+      } else {
+        cmd.p.type = 'normal'
+        cmd.p.name = req.url.replace('/pipelines/', '')
+      }
+      cmd.p.input = req.body
+
+      const body = {
+        ver: '9.0',
+        vars: { sid: sessionId },
+        cmds: [cmd]
+      }
+
+      const headers = {
+        'sg-application-id': `shop_${shopNumber}`,
+        'sg-device-id': deviceId
+      }
+      Object.assign(headers, defaultHeader)
+
+      const params = {
+        url: rapidUrl,
+        method: 'POST',
+        headers,
+        body,
+        json: true,
+        resolveWithFullResponse: true
+      }
+      return this._doRequest(params, res, next)
+    }
+  }
+
+  /**
+   * Check that the supplied arguments form a valid path in the object
+   * @param {Object} obj
+   * @param {...*} path
+   * @returns {boolean}
+   */
+  static _objectPathExists (obj, ...path) {
+    for (let i = 0; i < path.length; i++) {
+      if (!obj || !obj.hasOwnProperty(path[i])) {
+        return false
+      }
+      obj = obj[path[i]]
+    }
+    return true
+  }
+
+  /**
+   * Does actual proxy request to the rapid server
+   * @param {object} params
+   * @param {object} res
+   * @param next
+   */
+  _doRequest (params, res, next) {
```

Comment:
The response should just be the output of the pipeline response.

So instead of:
```javascript
{
    "cmds": [
        {
            "c": "pipelineResponse",
            "p": {
                "output": {
                    "categories": [
                    ]
                }
            }
        }
    ],
    "ver": "9.0"
}
```

this:
```
{
  "categories": []
}
```

Reply:
The same should be also in the error case